### PR TITLE
Document staged requests endpoints

### PR DESCRIPTION
**Pull request by saraycp**
**Reviewing Tips:**

- Open requests towards the original project (which has a staging workflow). You'll see those requests in the backlog.
- Use this endpoint to assign any of those requests to one of the staging projects.
- Play around with listing all the staged requests and "unstage" some of them too.

Controller: Staging::StagedRequestsController < Staging::StagingController

### CODE REVIEW

File: src/api/public/apidocs-new/paths/staging_project_name_staging_projects_staging_project_name_staged_requests.yaml [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+      content:
+        application/xml; charset=utf-8:
+          schema:
+            $ref: '../components/schemas/staging/staged_requests.yaml'
+          examples:
+            Multiple Requests:
+              value:
+                request:
+                  - id: 1
+                    creator: Iggy
```
Comment (danidoni) [inline src/api/public/apidocs-new/paths/staging_project_name_staging_projects_staging_project_name_staged_requests.yaml:20]: This response on my side looks like this:
```
<staged_requests>
  <request id="12" type="submit" creator="Admin" state="review" package="hello_world" superseded_by="" updated="2023-04-28T09:39:14Z"/>
</staged_requests>
```

File: src/api/public/apidocs-new/paths/staging_project_name_staging_projects_staging_project_name_staged_requests.yaml [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+      description: |
+        Bad Request.
+
+        XML Schema used for body validation: [status.xsd](../schema/status.xsd).
+      content:
+        application/xml; charset=utf-8:
+          schema:
+            $ref: '../components/schemas/api_response.yaml'
+          examples:
+            With Excluded Requsets:
```
Comment (danidoni) [inline src/api/public/apidocs-new/paths/staging_project_name_staging_projects_staging_project_name_staged_requests.yaml:90]:
```suggestion
            With Excluded Requests:
```

File: src/api/public/apidocs-new/paths/staging_project_name_staging_projects_staging_project_name_staged_requests.yaml [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    '400':
+      description: |
+        Bad Request.
+
+        XML Schema used for body validation: [status.xsd](../schema/status.xsd).
+      content:
+        application/xml; charset=utf-8:
+          schema:
+            $ref: '../components/schemas/api_response.yaml'
+          examples:
```
Comment (danidoni) [inline src/api/public/apidocs-new/paths/staging_project_name_staging_projects_staging_project_name_staged_requests.yaml:170]: Starting with the data that obs's `rake dev:test_data:create` creates, request #1 is not staged, if I try to remove that request, I get:
```
<status code="invalid_request">
  <summary>Error while unassigning requests: Request '1' is not staged and Requests with number: 2 don't belong to Staging: home:Admin</summary>
</status>
```

Comment (saraycp) [reply src/api/public/apidocs-new/paths/staging_project_name_staging_projects_staging_project_name_staged_requests.yaml:179]: They are two concatenated errors, I described them separately

File: src/api/public/apidocs-new/paths/staging_project_name_staging_projects_staging_project_name_staged_requests.yaml (hunk 2 above)
Comment (saraycp) [reply src/api/public/apidocs-new/paths/staging_project_name_staging_projects_staging_project_name_staged_requests.yaml]: corrected

File: src/api/public/apidocs-new/paths/staging_project_name_staging_projects_staging_project_name_staged_requests.yaml (hunk 1 above)
Comment (saraycp) [reply src/api/public/apidocs-new/paths/staging_project_name_staging_projects_staging_project_name_staged_requests.yaml:20]: True! I didn't complete the example... done!

**Review by danidoni — APPROVED** [review]

**Pull request merged**